src/params.rs

```diff
@@ -83,6 +83,7 @@ pub struct Params {
   pub inputs: Vec<Array>,
   pub outputs: Vec<Array>,
   pub recurrences: Vec<Array>,
+  pub deltaRec: Vec<Array>,
```

Comment:
You're right again. I used a vector because I was tired fighting the borrow checker and I wanted to submit it today. But yes, a simple reference to an Array is preferable.